Create 2023-state-of-rust-analysis.md (#1105)
rust-lang/blog.rust-lang.org: posts/2023-state-of-rust-analysis.md

Rust user respondents were asked which country they live in. The top 13 countries represented were as follows: United States (25%), Germany (12%), China (7%), United Kingdom (6%), France (5%), Canada (4%), Russia (4%), Japan (3%), Netherlands (3%), Sweden (2%), Australia (2%), Poland (2%), India (2%). Nearly 72.5% of respondents elected to answer this question. 

While we see global access to Rust education as a critical goal for our community, we are proud to say that Rust was used all over the world in 2022!

## Rust Usage
More people are using Rust than ever before! Of our respondents using Rust, 47% do so on a daily basis – an increase of 4% from the previous year. 

30% of Rust user respondents can write simple programs in Rust, 27% can write production-ready code, and 42% consider themselves productive using Rust. 

Over 90% of survey respondents identified as Rust users. Of the former Rust users who completed the survey, 30% cited difficulty as the primary reason for giving up while nearly 47% cited factors outside of their control. 

> ![1](https://github.com/rust-lang/blog.rust-lang.org/assets/45044840/54c56ffc-2a76-4144-9800-94f5e7da0180)

Reviewer: I think something might have gone wrong with these images since they are somehow not part of the pull request diff. I assume they should use relative paths to https://github.com/rust-lang/blog.rust-lang.org/tree/master/static/images?

@graciegregory let me know if you need help with that :)

---
layout: post
title: 2022 Annual Rust Survey Results
author: The Rust Survey Working Group in partnership with the Rust Foundation
release: false
---

Hello, Rustaceans!

> For the 6th year in a row, the Rust Project conducted a survey on the Rust programming language, with participation from project maintainers, contributors, and those generally interested in the future of Rust. This edition of the annual State of Rust Survey opened for submissions on December 5 and ran until December 22, 2022. Thank you for your patience as we’ve worked to assemble this analysis of the survey responses. We plan to identify a more expedient and sustainable process going forward. 

Reviewer 2: Is there a more detailed analysis of the data available? Specifically I was wondering whether there is a simple list of the questions + the aggregate results for each question on the github repo. If so, it'd be good to link to it.

Reviewer 3: I don't think we have that now. While it's certainly possible, I think it might take a bit of work to get into a format that is legible for general consumption. I wouldn't block this (long overdue) post on that, but we should try our best to get something like that out ASAP (cc @nrc).

> For the 6th year in a row, the Rust Project conducted a survey on the Rust programming language, with participation from project maintainers, contributors, and those generally interested in the future of Rust. This edition of the annual State of Rust Survey opened for submissions on December 5 and ran until December 22, 2022. Thank you for your patience as we’ve worked to assemble this analysis of the survey responses. We plan to identify a more expedient and sustainable process going forward. 

Reviewer 3: Perhaps we should add a link to the announcement blog post.

```suggestion
For the 6th year in a row, the Rust Project [conducted a survey on the Rust programming language](https://blog.rust-lang.org/2022/12/05/survey-launch.html), with participation from project maintainers, contributors, and those generally interested in the future of Rust. This edition of the annual State of Rust Survey opened for submissions on December 5 and ran until December 22, 2022. Thank you for your patience as we’ve worked to assemble this analysis of the survey responses. We plan to identify a more expedient and sustainable process going forward. 
```

The goal of this survey is always to give our wider community a chance to express their opinions about the language we all love and help shape its future. We’re grateful to those of you who took the time to share your voice on the state of Rust last year. 

Before diving into a few highlights, we would like to thank everyone who was involved in creating the State of Rust survey with special acknowledgment to the translators whose work allowed us to offer the survey in English, Simplified Chinese, Traditional Chinese, French, German, Japanese, Korean, Portuguese, Russian, Spanish, and Ukrainian. 

## Participation
In 2022, we had 9,433 total survey completions and an increased survey completion rate of 82% vs. 76% in 2021. While the goal is always total survey completion for all participants, the survey requires time, energy, and focus – we consider this figure quite high and were pleased by the increase. 

> 

Reviewer 3: ```suggestion
```

> We also saw a significant increase in the number of people viewing but not participating in the survey (from 16,457 views in 2021 to 2,5581 – a view increase of over 55%). While this is likely due to a number of different factors, we feel this information speaks to the rising interest in Rust and the growing general audience following its evolution. 

Reviewer 4: I assume:

```suggestion
We also saw a significant increase in the number of people viewing but not participating in the survey (from 16,457 views in 2021 to 25,581 – a view increase of over 55%). While this is likely due to a number of different factors, we feel this information speaks to the rising interest in Rust and the growing general audience following its evolution. 
```

We were pleased to offer the survey in 11 languages – more than ever before, with the addition of a Ukrainian translation in 2022. 77% of respondents took this year’s survey in English, 5% in Chinese (simplified), 4% in German and French, 2% in Japanese, Spanish, and Russian, and 1% in Chinese (traditional), Korean, Portuguese, and Ukrainian. This is our lowest percentage of respondents taking the survey in English to date, which is an exciting indication of the growing global nature of our community!

The vast majority of our respondents reported being most comfortable communicating on technical topics in English (93%), followed by Chinese (7%). 

Rust user respondents were asked which country they live in. The top 13 countries represented were as follows: United States (25%), Germany (12%), China (7%), United Kingdom (6%), France (5%), Canada (4%), Russia (4%), Japan (3%), Netherlands (3%), Sweden (2%), Australia (2%), Poland (2%), India (2%). Nearly 72.5% of respondents elected to answer this question. 

While we see global access to Rust education as a critical goal for our community, we are proud to say that Rust was used all over the world in 2022!

## Rust Usage

> More people are using Rust than ever before! Of our respondents using Rust, 47% do so on a daily basis – an increase of 4% from the previous year. 

Reviewer 2: Seems like yes -- I would have liked to see that overall number cited here.

![4](https://github.com/rust-lang/blog.rust-lang.org/assets/45044840/cb17b676-5904-4dbd-8cc9-31ee7aa94c7f)

Of those respondents that used Rust at work, 72% reported that it helped their team achieve its goals (a 4% increase from the previous year) and 75% have plans to continue using it on their teams in the future. 

But like any language being applied in the workplace, Rust’s learning curve is an important consideration; 39% of respondents using Rust in a professional capacity reported the process as “challenging” and 9% of respondents said that adopting Rust at work has “slowed down their team”. However, 60% of productive users felt Rust was worth the cost of adoption overall. 
![5](https://github.com/rust-lang/blog.rust-lang.org/assets/45044840/c533954b-eadc-41c0-ac9e-717d7ebc6e22)

It is exciting to see the continued growth of professional Rust usage and the confidence so many users feel in its performance, control, security and safety, enjoyability, and more!

## Supporting the Future of Rust

A key goal of the State of Rust survey is to shed light on challenges, concerns, and priorities Rustaceans are currently sitting with. 

> Of those respondents who shared their main worries for the future of Rust, 26% have concerns that the developers and maintainers behind Rust are not properly supported – a decrease of more than 30% from the previous year’s findings. While we would of course like to see this number at 0%, this improvement reflects the growing maturity of the Rust Foundation and the support they provided the Project in 2022 during its first full year in operation.

Reviewer 5: I'm not sure if this is the right channel to note this, but this statement:

> 26% have concerns that the developers and maintainers behind Rust are not properly supported – a decrease of more than 30% from the previous year’s findings.

Does not necessarily lead to this conclusion:

> this improvement reflects the growing maturity of the Rust Foundation and the support they provided the Project in 2022 during its first full year in operation.

While certainly the Foundation's involvement in supporting Rust maintainers and project members is important and shouldn't be understated, I was surprised by the wording the first time I read through this. There are other reasons why this number may have dropped -- for example, this is gauging sentiment from the Rust community at large, and not (if I'm understanding the data correctly) reporting the feelings of Rust project members and Rust maintainers themselves (who are a small percentage of this population as a whole). Also, project members and maintainers do receive other channels of support, such as partial or full compensation to work on Rust-related activities from employers and non-employer sponsorship such as Github sponsors and patreon.

Given all of that, I think this should be weakened to not make such a direct claim. Perhaps it's better to just use wording like "suggests", or back it up with other data that leads to this specific conclusion.
